### PR TITLE
lnprototests with CI

### PR DESCRIPTION
**Pull request by niftynei**
Let's run the lnprototests with every CI build, and make them easy to run while doing dev :hugs: 

In draft til I make sure all the CI dependency kinks are worked out

Changelog-None.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -105,7 +105,7 @@ jobs:
           NETWORK: ${{ matrix.NETWORK }}
           PYTEST_OPTS: ${{ matrix.PYTEST_OPTS }}
           PYTEST_PAR: ${{ matrix.PYTEST_PAR }}
-          TEST_CMD: ${{ matrix.TEST_CMD }}
+          TEST_CMD: "make check-protos"
```
Comment (niftynei) [inline .github/workflows/ci.yaml:108]: @cdecker what am i doing wrong here? It's set L72??

https://github.com/ElementsProject/lightning/pull/4444/commits/32ae95b1c7bfbeb7bf2d8213c84705df0746b2c1#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR72

Comment (niftynei) [reply .github/workflows/ci.yaml:108]: (oh without this change it runs the default `make pytest`

File: Makefile [hunk 2]
```diff
@@ -372,7 +372,14 @@ endif
 
 check-units:
 
-check: check-units installcheck pytest
+check: check-units installcheck pytest check-protos
+
+check-protos: $(ALL_PROGRAMS)
+ifeq ($(PYTEST),)
+	@echo "py.test is required to run the protocol tests, please install using 'pip3 install -r requirements.txt', and rerun 'configure'."
```
Comment (rustyrussell) [inline Makefile:379]: should this do `false` at the end?

File: requirements.txt [hunk 3]
```diff
@@ -4,36 +4,11 @@ https://github.com/ElementsProject/libwally-core/releases/download/release_0.8.0
...
-# Dependencies from pyln-proto
-base58 ~= 2.0.1
-bitstring ~= 3.1.6
-coincurve ~= 13.0.0
-cryptography ~= 3.2
-mypy ~= 0.790
-pysocks ~= 1.7.1
+./contrib/pyln-client
+./contrib/pyln-proto
+./contrib/pyln-testing
```
Comment (rustyrussell) [inline requirements.txt:12]: TIL this was a thing!

File: Makefile [hunk 4]
```diff
@@ -372,7 +372,7 @@ endif
 
 check-units:
 
-check: check-units installcheck pytest check-protos
+check: check-units installcheck check-protos pytest
```
Comment (rustyrussell) [inline Makefile]: This commit is a noop?

File: contrib/pyln-spec/setup.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
+from setuptools import setup
```
Comment (rustyrussell) [inline contrib/pyln-spec/setup.py:1]: I just want to make sure you know *why* the "../setup.py" in the sub- setup.py files works.

It reflects a great deal of, um, opreational creativity?

**Review by rustyrussell — APPROVED** [review]
Minor comments only: there seems to be a semantically-empty commit there, too.

File: contrib/pyln-spec/setup.py (hunk 5 above)
Comment (niftynei) [reply contrib/pyln-spec/setup.py:1]: @cdecker i'm deferring this one to you :)

File: Makefile (hunk 2 above)
Comment (niftynei) [reply Makefile:379]: I don't think I quite get your meaning here.

Comment (niftynei) [reply Makefile]: Ah I see, I missed the `exit 1` from where i copy-pasta'd this from :facepalm:

File: Makefile (hunk 4 above)
Comment (niftynei) [reply Makefile]: it is! removed.

**Review by rustyrussell — APPROVED** [review]
Ack https://github.com/ElementsProject/lightning/commit/dacceadf326c231fe6da0a5b775eeebd0d201f2f

**Pull request merged**